lib/metrics/eventLoopLag.js

```diff
@@ -37,32 +37,35 @@ module.exports = (registry, config = {}) => {
 	const labelNames = Object.keys(labels);
 	const registers = registry ? [registry] : undefined;
 
-	let collect;
-	if (!perf_hooks || !perf_hooks.monitorEventLoopDelay) {
-		collect = () => {
-			const start = process.hrtime();
-			setImmediate(reportEventloopLag, start, lag, labels);
-		};
-	} else {
-		const histogram = perf_hooks.monitorEventLoopDelay({
-			resolution: config.eventLoopMonitoringPrecision,
-		});
-		histogram.enable();
+	let collect = () => {
+		const start = process.hrtime();
+		setImmediate(reportEventloopLag, start, lag, labels);
+	};
 
-		collect = () => {
-			const start = process.hrtime();
-			setImmediate(reportEventloopLag, start, lag, labels);
+	if (perf_hooks && perf_hooks.monitorEventLoopDelay) {
+		try {
+			const histogram = perf_hooks.monitorEventLoopDelay({
+				resolution: config.eventLoopMonitoringPrecision,
+			});
+			histogram.enable();
 
-			lagMin.set(labels, histogram.min / 1e9);
-			lagMax.set(labels, histogram.max / 1e9);
-			lagMean.set(labels, histogram.mean / 1e9);
-			lagStddev.set(labels, histogram.stddev / 1e9);
-			lagP50.set(labels, histogram.percentile(50) / 1e9);
-			lagP90.set(labels, histogram.percentile(90) / 1e9);
-			lagP99.set(labels, histogram.percentile(99) / 1e9);
+			collect = () => {
+				const start = process.hrtime();
+				setImmediate(reportEventloopLag, start, lag, labels);
 
-			histogram.reset();
-		};
+				lagMin.set(labels, histogram.min / 1e9);
+				lagMax.set(labels, histogram.max / 1e9);
+				lagMean.set(labels, histogram.mean / 1e9);
+				lagStddev.set(labels, histogram.stddev / 1e9);
+				lagP50.set(labels, histogram.percentile(50) / 1e9);
+				lagP90.set(labels, histogram.percentile(90) / 1e9);
+				lagP99.set(labels, histogram.percentile(99) / 1e9);
+
+				histogram.reset();
+			};
+		} catch {
+			// bun has `perf_hooks.monitorEventLoopDelay` defined but throws "NotImplemented" when called
+		}
```

Comment:
the `catch` should check for the specific error bun throws and rethrow any other